lib/archethic_web/live/chains/beacon_live.ex

```diff
@@ -0,0 +1,133 @@
+defmodule ArchEthicWeb.BeaconChainLive do
+  @moduledoc false
+  use ArchEthicWeb, :live_view
+
+  alias ArchEthic.TransactionChain.{
+    Transaction,
+    TransactionData
+  }
+
+  alias ArchEthic.BeaconChain
+  alias ArchEthic.BeaconChain.Summary, as: BeaconSummary
+  alias ArchEthic.BeaconChain.SummaryTimer
+  alias ArchEthic.Crypto
+  alias ArchEthic.P2P
+  alias ArchEthic.P2P.Node
+  # alias ArchEthic.PubSub
+  alias ArchEthic.SelfRepair.Sync.BeaconSummaryHandler
+  alias ArchEthicWeb.ExplorerView
+  alias Phoenix.View
+
+  def fetch_txn_summaries(date1 = %DateTime{}, date2 = %DateTime{}) do
+    summary_pools = BeaconChain.get_summary_pool(date1, date2)
+
+    Enum.map(summary_pools, fn {subset, nodes_by_summary_time} ->
+      Enum.map(nodes_by_summary_time, fn {summary_time, nodes} ->
+        {nodes, subset, summary_time}
+      end)
+    end)
+    |> :lists.flatten()
+    |> Task.async_stream(
+      fn {nodes, subset, summary_time} ->
+        beacon_address = Crypto.derive_beacon_chain_address(subset, summary_time, true)
+        %Node{network_patch: patch} = P2P.get_node_info()
+
+        beacon_address
+        |> BeaconSummaryHandler.download_summary(nodes, patch)
+      end,
+      on_timeout: :kill_task,
+      max_concurrency: 256
+    )
+    |> Enum.filter(&match?({:ok, {:ok, %Transaction{}}}, &1))
+    |> Enum.map(fn {:ok, {:ok, %Transaction{data: %TransactionData{content: content}}}} ->
+      {summary, _} = BeaconSummary.deserialize(content)
+      %BeaconSummary{transaction_summaries: transaction_summaries} = summary
+      transaction_summaries
+    end)
+  end
+
+  def mount(_params, _session, socket) do
+    # Todo handle live transactions
+    # if connected?(socket) do
+    #   PubSub.register_to_new_transaction_by_type(:beacon)
+    #   PubSub.register_to_new_transaction_by_type(:beacon_summary)
+    # end
+
+    beacon_dates = get_beacon_dates()
+
+    new_assign =
+      socket
+      |> assign(:dates, beacon_dates)
+      |> assign(:current_date_page, 1)
+      |> assign(
+        :transactions,
+        list_transactions_by_date(
+          Enum.at(beacon_dates, 0),
+          fetch_txn_summaries(Enum.at(beacon_dates, 0), Enum.at(beacon_dates, 1))
+        )
+      )
+
+    {:ok, new_assign}
+  end
+
+  def render(assigns) do
+    View.render(ExplorerView, "beacon_chain_index.html", assigns)
+  end
+
+  def handle_params(%{"page" => page}, _uri, socket = %{assigns: %{dates: dates}}) do
+    case Integer.parse(page) do
+      {number, ""} when number > 0 ->
+        transactions =
+          list_transactions_by_date(
+            Enum.at(dates, number + 1),
```

Comment:
You should avoid duplicated calls to a Enum, as it will iterate twice over the same list

Reply:
so, instead of calling Enum 3 times I should use two var which will store them and pass them to fetch. Right this will  reduce Enum calls from 3 to 2

Reply:
If you use fetch_txs_summaries with a single date it should be better I think

Reply:
I am passing two dates to BeaconChain.get_summary_pool/2 as BeaconChain.get_summary_pools/1 takes one date and get data from the last sync date. So, I am trying a get that data in particular time interval say date1 and date2.

Reply:
Summary transaction address are based on a single date. So I don't see yet the need to use two dates to get the transaction.
Moreover it should improve the performance I think.